### DISCUSSION TITLE
Unable to include Content-Type header

### DISCUSSION THREAD

**Discussion by patwhitehead1** [soto-project/soto#676] [Q&A]
I'm trying to create a pre-signed url for uploading to S3. I can successfully upload an image file, but the type is not visible once it's in S3. When I download it from S3, the file appears to be corrupted. I believe this has to do with setting the `Content-Type`. Here's my code for generating the pre-signed URL:

`let signedURL = try await s3.signURL(url: url, httpMethod: method, headers: ["Content-Type": "image/jpeg"], expires: .minutes(15))`

After generating, this is the only value in signed headers: `SignedHeaders=host`.

Any idea what's going on?

**Comment by patwhitehead1** [comment]
Now that I look closer, `Content-Type` is intentionally removed when `SigningData` is created, so I think my problem lies elsewhere.

**Reply by adam-fowler** [reply]
This could be considered a bug content-type is removed because it is in the list of headers that don't get signed. But if you included it initially the list of returned headers should include it. 

Have you tried adding the header back in to see what results you get?

**Reply by adam-fowler** [reply]
Actually ignore the last comment, we are good.

**Reply by adam-fowler** [reply]
I found this in the docs 
```
If the Content-Type header is present in the request, you must add it to the CanonicalHeaders list.
```
So I'm going to stop removing it from the list of signed headers https://github.com/soto-project/soto-core/pull/561

**Reply by patwhitehead1** [reply]
Thanks for looking into this. It turns out the issue was with how I was including the image data in the request body, it wasn't related to the presigned url.
